### DISCUSSION TITLE
Tuya Cloud OpenAPI - permission denied

### DISCUSSION THREAD

**Discussion by BenoitVH** [jasonacox/tinytuya#329] [Q&A]
Hi, 

I've been using TinyTuya for quite some months now (and very happy that Jason provided this awesome tool :).
Every now and then I have launched "TinyTuya wizard" to get the keys when I add new devices. 
This has worked fine for dozens of times but today however I received an error "0 registered devices"...
When I enabled "debugging" this points to a "permission denied" message.
From the log file below, one can see that initial connection was successful but getting device data (with the same credentials) isn't.

Has anyone come across this issue; perhaps Tuya has (once again) changed their API ?

I assume I can bypass the issue by starting completely from scratch by creating a new "Tuya cloud project"; has anybody tried that before (and are there any side-effects to consider) ?

Many thanks in advance!

Benoit

Debug output (python -m tinytuya wizard -debug)
=============

DEBUG:Starting new HTTPS connection (1): openapi.tuyaeu.com:443
DEBUG:https://openapi.tuyaeu.com:443 "GET /v1.0/token?grant_type=1 HTTP/1.1" 200 235
DEBUG:GET: URL=https://openapi.tuyaeu.com/v1.0/token?grant_type=1 HEADERS={'secret': '49ee6e415fb74d75a0be43f9177fe7d5', 'client_id': '5dy44y7n33arv7ngwepu', 'sign': '5B9427494501011DF0A90677396FE1C0EC4D684CA4EEEBED6EA4A41ADA4F1D2E', 't': '1681577786066', 'sign_method': 'HMAC-SHA256'} response code=200 text={"result":{"access_token":"30b227dff574bc7912271b06836191dc","expire_time":5974,"refresh_token":"519a1bb70c75b74452f6bbc1c6112e1d","uid":"bay1671210199361OQS3"},"success":true,"t":1681577786256,"tid":"759d7d7fdbae11edbe55ce1a38a7d30c"} token=None
DEBUG:server_time_offset: 0
DEBUG:Starting new HTTPS connection (1): openapi.tuyaeu.com:443
DEBUG:https://openapi.tuyaeu.com:443 "GET /v1.0/iot-01/associated-users/devices?size=50 HTTP/1.1" 200 112
DEBUG:GET: URL=https://openapi.tuyaeu.com/v1.0/iot-01/associated-users/devices?size=50 HEADERS={'client_id': '5dy44y7n33arv7ngwepu', 'sign': '0B4C10B12E0278A2D90F191036A7F096E718AF9423E44D059F86B0DEDF4C50DF', 't': '1681577786279', 'sign_method': 'HMAC-SHA256', 'access_token': '30b227dff574bc7912271b06836191dc'} response code=200 text={"code":1106,"msg":"permission deny","success":false,"t":1681577786429,"tid":"75b694d2dbae11edb3c5c22eba99f181"} token=30b227dff574bc7912271b06836191dc
DEBUG:Cloud response:
DEBUG:{
  "code": 1106,
  "msg": "permission deny",
  "success": false,
  "t": 1681577786429,
  "tid": "75b694d2dbae11edb3c5c22eba99f181"
}

**Comment by uzlonewolf** [comment]
Same issue as #323, fixed in v1.12.3.
